### PR TITLE
Additional /auditLog methods for selecting log entries

### PR DESCRIPTION
**Pull request by tsschmidt**
PR adds the following features for the /auditLog endpoint:
 - Retrieve log entries from server for a period of time starting from now.  Resolution can be from from Months to Seconds specified as duration (PT1H)
 - Filter Log entries by the following:
    - Action Performed
    - Principal
    - Client IP
    - Resource Operated on.

### CODE REVIEW

**Comment by mmoayyed** [comment]
Looks good. Update the docs please to note the new endpoint behavior.

File: docs/cas-server-documentation/installation/Audits.md [hunk 1]
```diff
@@ -23,6 +23,25 @@ The following endpoints are provided by CAS:
 |--------------------------|------------------------------------------------
 | `auditLog`               | Provides a JSON representation of all the audit log.
 
+Navigating to `/actuator/auditLog` displays the complete audit for the past number of days determined by `cas.audit.numberOfDaysInHistory`(default is 30 days);
+
+You can specify an interval of log entries to return by adding a Duration Syntax to the navigated path.  This interval will be subtracted from the current 
+date and time when the query is executed.  For instance `/actuator/auditLog/PT1H` will return only entries for the past our.  An additional path can be added to 
+limit the types of entries that will be returned (i.e `/actuator/auditLog/PTIH/SERVICE_TICKET_CREATED`).
```
Comment (mmoayyed) [inline docs/cas-server-documentation/installation/Audits.md]: These should not be selectors. There are query parameters.

Comment (tsschmidt) [reply docs/cas-server-documentation/installation/Audits.md]: They don't work as query params though, they do as path selectors.

Comment (mmoayyed) [reply docs/cas-server-documentation/installation/Audits.md]: Why not?

Comment (tsschmidt) [reply docs/cas-server-documentation/installation/Audits.md]: Sorry, it was my misunderstanding.

**Pull request merged**